### ISSUE TITLE
reportGeneralTypeIssues doesn't accept operator "*" for str and int 

### ISSUE THREAD

**Issue by garrickf** [microsoft/pyright#1373]
This might be better suited for [typeshed](https://github.com/python/typeshed/issues), but I'm not sure, so I thought I'd start here.

**Describe the bug**
Upon multiplying a string and int in Python (to repeat the string), pyright claims `Operator "*" not supported for types "str" and "float" Pylance(reportGeneralTypeIssues)`

**To Reproduce**

1. Have these user settings in VS Code:

```
    "python.languageServer": "Pylance",
    "python.analysis.typeCheckingMode": "basic",
```

2. Attempt to multiply a string and an int

**Expected behavior**
I would expect the message to not pop up, as the operation is supported (unlike other operators between str and int, like '+'—in which case the error message is useful. Is the message suggesting this is not good Python style in general or not "good style" to a type checker?

**Code**
```python
s1 = str("?")
s2 = s1 * 8
```

**VS Code extension or command-line**
Running as a VS Code (1.52.1) extension, Pylance 2021.1.1

**Additional context**
N/A

**Comment by jakebailey** [comment]
This is explicitly supported: https://github.com/python/typeshed/blob/c893857f4671e454d558c3fcaf1a502548669873/stdlib/3/builtins.pyi#L396

And with the same setup it seems to work for me.

![image](https://user-images.githubusercontent.com/5341706/104661375-43016e00-567d-11eb-94e5-8c72d0ce1094.png)

Is that the full code sample, or did this appear elsewhere?

**Comment by garrickf** [comment]
Oh, I'm silly—in my code example (which I simplified for the issue) it's a float:

```
message = s1 * (n / 2)
```

Should be `//`, can't multiply a string and float. I guess pyright was helping me out here!
